compositor_render/src/transformations/image_renderer.rs

```diff
@@ -0,0 +1,260 @@
+use std::{
+    str::{from_utf8, Utf8Error},
+    sync::{Arc, Mutex},
+};
+
+use bytes::{Bytes, BytesMut};
+use compositor_common::{scene::Resolution, transformation::ImageSpec};
+use image::ImageFormat;
+use resvg::{
+    tiny_skia,
+    usvg::{self, TreeParsing},
+};
+
+use crate::renderer::{
+    texture::{NodeTexture, RGBATexture},
+    RegisterTransformationCtx, RenderCtx, WgpuCtx,
+};
+
+pub struct Image {
+    renderer: ImageRenderer,
+}
+
+impl Image {
+    pub fn new(ctx: &RegisterTransformationCtx, spec: ImageSpec) -> Result<Self, ImageError> {
+        let file = Self::download_file(spec.url())?;
+        let renderer = match spec {
+            ImageSpec::Png { .. } => {
+                let asset = BitmapAsset::new(&ctx.wgpu_ctx, file, ImageFormat::Png)?;
+                ImageRenderer::BitmapAsset(asset)
+            }
+            ImageSpec::Jpeg { .. } => {
+                let asset = BitmapAsset::new(&ctx.wgpu_ctx, file, ImageFormat::Jpeg)?;
+                ImageRenderer::BitmapAsset(asset)
+            }
+            ImageSpec::Svg { resolution, .. } => {
+                let asset = SVGAsset::new(&ctx.wgpu_ctx, file, resolution)?;
+                ImageRenderer::Svg(asset)
+            }
+            ImageSpec::Gif { .. } => {
+                // TODO: support dynamic GIFs
+                let asset = BitmapAsset::new(&ctx.wgpu_ctx, file, ImageFormat::Gif)?;
+                ImageRenderer::BitmapAsset(asset)
+            }
+        };
+        Ok(Self { renderer })
+    }
+
+    fn download_file(url: &str) -> Result<bytes::Bytes, ImageError> {
+        // TODO: support local files
+        let response = reqwest::blocking::get(url)?;
+        let response = response.error_for_status()?;
+        Ok(response.bytes()?)
+    }
+}
+
+pub struct ImageNode {
```

Comment:
This struct is not needed, but we will want to support images that are not just a static bitmap e.g.
- GIF with animations
- SVG with resolution specified at the update scene level, and not during the transformation registration)

Reply:
outdated